src/refspec.c

```diff
@@ -324,7 +324,7 @@ int git_refspec__dwim_one(git_vector *out, git_refspec *spec, git_vector *refs)
 		for (j = 0; formatters[j]; j++) {
 			git_buf_clear(&buf);
 			if (git_buf_printf(&buf, formatters[j], spec->src) < 0)
-				return -1;
+				goto out_err;
```

Comment:
Similar to the changes in `rebase.c`, we could have a `GITERR_CHECK_ALLOC_BUF` or something like that to check a `git_buf` for OOM status.  Or something.